sdk/servicebus/service-bus/test/perf/rhea-promise/send.ts

```diff
@@ -85,16 +86,24 @@ async function RunTest(
   await connection.close();
 }
 
-async function ExecuteSendsAsync(sender: any, messages: number): Promise<void> {
+async function ExecuteSendsAsync(sender: Sender, messages: number): Promise<void> {
+  sender.setMaxListeners(1000);
+  const onSuccessOrFail: Func<EventContext, void> = (context: EventContext) => {
+    canSend = true;
+    sender.removeListener(SenderEvents.rejected, onSuccessOrFail);
+    sender.removeListener(SenderEvents.accepted, onSuccessOrFail);
+  };
   while (++_messages <= messages) {
-    while (!sender.sendable()) {
+    while (!sender.sendable() || canSend === false) {
       await delay(0.01);
```

Comment:
It may not matter, but I would decrease this delay to say `.001`.  If this delay is too large, I believe it could become a bottleneck in the client.

Reply:
I did verify 0.001 and 0.0001(long back, when I had written this sample for the first time).

My earlier observations
- when set to 0.001, the `sendable()` check is being hit hundreds of times(which felt redundant)
- when set to 0.01, the number of times the `sendable()` check is being hit reduced to multiples of ten(which is better than 0.001 case)

So, settled with 0.01(but this is with standard namespace).
Any thoughts ?

Reply:
I assumed the time was in seconds (meaning .01s is 10ms), but I realized the time is in ms.  So this time might actually be too low which causes wasted cycles spinning in this loop.  I'll try 1ms and see what happens.

I also think the test can be changed to remove the polling entirely, and instead send the next message when the previous message event is raised.

Reply:
>I also think the test can be changed to remove the polling entirely, and instead send the next message when the previous message event is raised.

I agree. 
The `sendable()` check ensures that we send messages only when we have the credits.
Adding this check avoided the problem of not being able to send the messages once the limit is reached.
Now that we are having event listeners in the sample, we can try invoking the send if the accepted or rejected event is observed.